includes/connect/class-wc-stripe-connect.php

```diff
@@ -396,5 +499,82 @@ public function refresh_connection() {
 			// save_stripe_keys() schedules a connection_refresh after saving the keys,
 			// we don't need to do it explicitly here.
 		}
+
+		/**
+		 * Redacts sensitive information from strings, arrays, or objects.
+		 *
+		 * @param string|array|object $data The string, array, or object to redact sensitive information from.
+		 * @return string|array
+		 */
+		public static function redact_sensitive_data( $data ) {
+			$sensitive_keys = [
+				'_wpnonce',
+				'state',
+				'code',
+				'secretKey',
+				'refreshToken',
+				'secret_key',
+				'test_secret_key',
+				'webhook_secret',
+				'test_webhook_secret',
+			];
+
+			if ( is_object( $data ) ) {
+				// Handle objects (stdClass) by converting to array, processing, and returning as array.
+				$data = (array) $data;
+			}
+
+			if ( is_array( $data ) ) {
+				// Handle arrays recursively
+				$redacted = [];
+				foreach ( $data as $key => $value ) {
+					if ( in_array( $key, $sensitive_keys, true ) && is_string( $value ) && ! empty( $value ) ) {
+						$redacted[ $key ] = self::redact_string( $value );
+					} else {
+						$redacted[ $key ] = self::redact_sensitive_data( $value );
+					}
+				}
+				return $redacted;
+			}
+
+			if ( is_string( $data ) ) {
+				// Handle a form-urlencoded string (like an URI or form payload).
+				foreach ( $sensitive_keys as $key ) {
+					$data = preg_replace_callback(
+						'/([?&]' . preg_quote( $key, '/' ) . '=)([^&#]*)/i',
+						function ( $matches ) {
+							$value = $matches[2];
+							if ( strlen( $value ) > 0 ) {
+								return $matches[1] . self::redact_string( $value );
+							}
+							return $matches[0];
+						},
+						$data
+					);
```

Comment:
Not blocking: Is it worth adding an early return if we don't have a URL or form payload so we're not running multiple unnecessary regular expression matches?
```suggestion
				if ( ! str_contains( $data, '?' ) && ! str_contains( $data, '&' ) ) {
					return $data;
				}
				foreach ( $sensitive_keys as $key ) {
					$data = preg_replace_callback(
						'/([?&]' . preg_quote( $key, '/' ) . '=)([^&#]*)/i',
						function ( $matches ) {
							$value = $matches[2];
							if ( strlen( $value ) > 0 ) {
								return $matches[1] . self::redact_string( $value );
							}
							return $matches[0];
						},
						$data
					);
```

Reply:
It's also be used to redact the POST body (`application/x-www-form-urlencoded`) in the call to exchange the auth code, which sends only the `code` param (no `?` or `&`)
```
code=a188926...6f4a3b
```